packages/peregrine/src/Toasts/useToasts.js

```diff
@@ -116,16 +117,28 @@ export const useToasts = () => {
             const handleAction = () =>
                 onAction ? onAction(() => removeToast(id)) : () => {};
 
-            // A timeout of 0 means no auto-dismiss.
+            const pauseToast = () => {
+                allowToastToDie = false;
```

Comment:
This value will be lost on a re-render. We shouldn't use variables outside of the callback's scope.